src/ride/controllers/search_ride_controller.py

```diff
@@ -0,0 +1,29 @@
+from api.utils import failure_response
+from api.utils import success_response
+from api.utils import update
+from django.core.exceptions import ObjectDoesNotExist
+from person.models import Person
+
+from ..models import Ride
+from ..models import Path
+
+class SearchRideController:
+    def __init__(self, data, request, serializer): 
+        self._request = request
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        departure_datetime = self._data.get("departure_datetime")
+        start_location_name = self._data.get("start_location_name")
+        end_location_name = self._data.get("end_location_name")
+        if not (Path.objects.filter(start_location_name = start_location_name, end_location_name = end_location_name)).exists():
+            return failure_response("Path does not exist.")
+        
+        path = Path.objects.get(start_location_name = start_location_name, end_location_name = end_location_name)
+
+        if not (Ride.objects.filter(path = path)):
+            return failure_response("Ride with that path does not exist")
+        all_rides = Ride.objects.filter(path = path)
```

Comment:
we shouldn't just return rides with the desired path; we also want to make sure the ride's departure time is the same / similar to the departure_datetime given in the request body

ideally we will get return all rides 24 hours before and after the given datetime. the documentation link above should help with doing this!

Reply:
+1 hard filtering for rides with desired paths vs soft filtering with paths that are similar (around the same time) is important especially given the context of this app since more often than not we'll be showing similar paths